### ISSUE TITLE
[enhancement] retry sources download when their checksum fail

### ISSUE THREAD

**Issue by mattia-b89** [archlinuxfr/yaourt#311]
Sometime happens downloaded files are corrupted, mostly due to network issues, this stops yaourt from compiling the package.

I'd like to have:
yaourt automatically try to re-download missing sources (I think the times yaourt can do it is three, or even better, this will be a new option defined in yaourtrc)
OR
yaourt instead of simply ask for for a retry, a new whole compilation, give you the option to delete corrupted/not-passing check sources, prior to try the compilation

**Comment by f2404** [comment]
Doesn't it already prompt to retry building a package after a build failure has happened?

**Comment by mattia-b89** [comment]
Yes, Yaourt prompts if you want to start gain the building process.
I you say yes, the process re-start, Yaourt does not download sources again
because they are already on the disk and thus the process fails because one
(or more) sources do not pass the integrity check
It is like an infinite-loop...!

**Comment by f2404** [comment]
Ah, I can see the point now, thanks.
